seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/sink/KafkaSinkFactory.java

```diff
@@ -39,23 +39,26 @@ public String factoryIdentifier() {
     @Override
     public OptionRule optionRule() {
         return OptionRule.builder()
-                .required(Config.FORMAT, Config.BOOTSTRAP_SERVERS)
-                .conditional(
-                        Config.FORMAT,
-                        Arrays.asList(
-                                MessageFormat.JSON,
-                                MessageFormat.CANAL_JSON,
-                                MessageFormat.TEXT,
-                                MessageFormat.OGG_JSON,
-                                MessageFormat.AVRO),
-                        Config.TOPIC)
+                .required(Config.TOPIC, Config.BOOTSTRAP_SERVERS)
                 .optional(
+                        Config.FORMAT,
                         Config.KAFKA_CONFIG,
                         Config.ASSIGN_PARTITIONS,
                         Config.TRANSACTION_PREFIX,
                         Config.SEMANTICS,
                         Config.PARTITION,
                         Config.PARTITION_KEY_FIELDS)
+                .conditional(
```

Comment:
why `requiredOptions` is null?

Reply:
move topic to required check.

Reply:
So as you said, I think we should remove this conditional.

Reply:
Sorry, I made a mistake. I have reviewed the code for seatunnel-web again and can delete the option.

Reply:
removed. 
I have some question about the `optional` rule. we know if in `required`, then we must pass the parameter. if in `optional`, we can pass or not pass this parameter. 
But we still can pass some unknown parameter to the config, like this
```hacon
sink {
  kafka {
     topic = ""
     bootstrap.servers = ""
     k1 = v1
     k2 = v2
  }
}
```
i pass 2 parameter k1 and k2. it is not in option list and won't be use in feature. 
So what's the `option` rule value? it can't control anything

Reply:
It can verify the legality of the value corresponding to a meaningful key, thereby ensuring the normal operation of the job, and that unexpected values can be fed back to the user as early as possible. But for useless keys, not telling users to delete them is based on two principles: 1. These keys will not affect the normal operation of the job. 2. If we delete a key, the key may still exist in the user's own config. We hope that the user can run job normally without changing the config.